### PR TITLE
types: improve CoreAPI error handling

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Rename `CoreAPIErrorResponse` to `CoreAPIError`
- Stricter parsing of `core` responses with more logging

## Risk

Minor risk of messing with front/core with the added parsing but low.

## Deploy Plan

- deploy `front`

### CODE REVIEW

**Review by philipperolet — APPROVED** [review]
LGTM :+1: with 2 coms, addressing optional

File: types/src/front/lib/core_api.ts [hunk 1]
```diff
@@ -29,11 +29,23 @@ export const EMBEDDING_CONFIG = {
...
 };
 
-export type CoreAPIErrorResponse = {
+export type CoreAPIError = {
   message: string;
   code: string;
 };
-export type CoreAPIResponse<T> = Result<T, CoreAPIErrorResponse>;
+
+export function isCoreAPIError(obj: unknown): obj is CoreAPIError {
```
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/core_api.ts:37]: We don't have codes list in front for now so it has to be wide. But that's fine the goal is to check that something we believe is a core error is indeed one. Nothing else, right?

File: types/src/front/lib/core_api.ts [hunk 2]
```diff
@@ -1058,23 +1070,77 @@ export class CoreAPI {
...
+            message: "Unexpected response format from CoreAPI",
+          };
+          this._logger.error(
+            { error: err, json, status: response.status },
+            "CoreAPI error"
+          );
+          return new Err(err);
+        }
+      } catch (e) {
+        return parseError(response, e as SyntaxError);
```
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/core_api.ts:1142]: fair, will updae, nothing throws outside of this.

[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/core_api.ts:1142]: the problem is then you have to have a variable before you parse outside of the try catch etc...

will got with current form factor

[the remark this answers is not in the document]
Comment (philipperolet) [reply types/src/front/lib/core_api.ts:1142]: Was worth it imho but also fine to keep as ois

**Pull request merged**